### PR TITLE
feat: return structured mutation and insertion responses

### LINKED ISSUE #710: insertion not in orderBy fields for nucleotideInsertions and aminoAcidInsertions
When I try ordering insertions by their name like this:
[https://lapis.cov-spectrum.org/gisaid/v2/sample/aminoAcidInsertions?accessKey={accessKey}&orderBy=insertion&limit=3](https://lapis.cov-spectrum.org/gisaid/v2/sample/aminoAcidInsertions?accessKey={accessKey}&orderBy=insertion&limit=3), 
I get `"Error from SILO: OrderByField insertion is not contained in the result of this operation."`

### LINKED ISSUE #718: Return mutations and insertions as structured data
SILO now returns structured data for insertions and mutations. This should now also be returned by LAPIS. The SILO PR on insertions https://github.com/GenSpectrum/LAPIS-SILO/pull/373 waits on this issue to be merged. The mutations PR is already merged (https://github.com/GenSpectrum/LAPIS-SILO/pull/357).

The data looks like this:
Mutations: 
```
{
      "count": 37,
      "mutation": "T19R",
      "mutationFrom": "T",
      "mutationTo": "R",
      "position": 19,
      "proportion": 0.38144329896907214,
      "sequenceName": "S"
 }
```

Insertions:
```
{
      "count": 5,
      "inserted_symbols": "EPE",
      "insertion": "ins_S:214:EPE",
      "position": 214,
      "sequenceName": "S"
    },
```

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #718
resolves #710 

builds on https://github.com/GenSpectrum/LAPIS-SILO/pull/373

## PR Checklist
- [x] All necessary documentation has been adapted.
- [x] The implemented feature is covered by an appropriate test.

### CODE REVIEW

File: lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloQueryModel.kt [hunk 1]
```diff
@@ -50,13 +50,23 @@ class SiloQueryModel(
             ),
         )
         return data.map {
-            val sequenceName =
-                if (referenceGenomeSchema.isSingleSegmented()) it.mutation else "${it.sequenceName}:${it.mutation}"
+            val mutation = if (referenceGenomeSchema.isSingleSegmented()) {
+                it.mutation
+            } else {
+                "${it.sequenceName}:${it.mutation}"
+            }
```
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloQueryModel.kt:57]: @Taepper should we move this logic to SILO, too?

File: lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloQueryModel.kt [hunk 2]
```diff
@@ -50,13 +50,23 @@ class SiloQueryModel(
...
 
             NucleotideMutationResponse(
-                sequenceName,
-                it.count,
-                it.proportion,
+                mutation = mutation,
+                count = it.count,
+                proportion = it.proportion,
+                sequenceName = when (referenceGenomeSchema.isSingleSegmented()) {
+                    true -> null
```
Comment (JonasKellerer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloQueryModel.kt:64]: Is null correct here? Do we want to set it to that specific value?

**Review by JonasKellerer — APPROVED** [review]
LGTM

I think the docu needs to be adapted at `/concepts/response-format/`. There we have a picture of the mutation endpoint swagger ui. 
And also the python example in `getting-started/generate-your-request`

File: lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloQueryModel.kt (hunk 2 above)
Comment (fengelniederhammer) [reply lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloQueryModel.kt:64]: We want to specifically set `null`.

**Pull request merged**